pkg/crd/crdvalidations.go

```diff
@@ -53,17 +53,61 @@ var (
 		},
 	}
 
+	// Function validation schema properties
+	functionSchemaPropsV2 = map[string]apiextensionsv1beta1.JSONSchemaProps{
```

Comment:
Overall changes look ok but I have a few points which we can discuss. I will also try this out with some scenarios and then we can close!